File: 프로그래머스/0/120904. 숫자 찾기/숫자 찾기.py
def solution(num, k):
    answer = str(num).find(str(k))  # 문자열로 변환한 num에서 k를 찾아 인덱스를 저장

    if answer > -1:  # k를 찾은 경우
        answer += 1  # 인덱스는 0부터 시작하므로 1을 더한다.

    return answer  # 최종 결과 반환

# 테스트 예제 실행
print(solution(29183, 1))  # 3
print(solution(232443, 4))  # 4
print(solution(123456, 7))  # -1

File: 프로그래머스/0/120904. 숫자 찾기/숫자 찾기.java
class Solution {
    public int solution(int num, int k) {
        String numStr = String.valueOf(num); // num을 문자열로 변환
        int index = numStr.indexOf(String.valueOf(k)); // k의 위치 찾기

        if (index != -1) {
            return index + 1; // 0-based index를 1-based로 변환하여 반환
        } else {
            return -1; // k가 포함되지 않으면 -1 반환
        }
    }

    // 테스트 예제 실행
    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.solution(29183, 1)); // 3
        System.out.println(sol.solution(232443, 4)); // 4
        System.out.println(sol.solution(123456, 7)); // -1
    }
}
